### PR TITLE
Rubocop step2

### PR DESCRIPTION
**Pull request by ineverov**
This PR is #256 with merged #265 and additional rubocop fixes for elements_container and tests

### CODE REVIEW

**Comment by luke-hill** [comment]
After merging in your second PR (At which point this becomes 1 commit). I think it would be better if instead of nuking all of the rubocop overrides we did them in smaller chunks.

I think some of the code you've "fixed" up becomes harder to read because of rubocop. I'm always a fan of easier to read code than more conformative code.

Maybe do the line length one in its entirety in one PR, or do it on a file by file basis. Just so it's a bit easier to review.

**Comment by luke-hill** [comment]
This can now be rebased into a single commit. But from that ideally we need to try split things out into a couple of PRs.

**Comment by ineverov** [comment]
@luke-hill I rebased this branch. Do you still want to split it into several commits?

**Comment by luke-hill** [comment]
Lets split it (If possible) The `LineLength` one is the main one. So that on it's own.

Alternatively we can leave this as is. But it'll be a heck of a thing to review that's all.

File: features/step_definitions/element_steps.rb [hunk 1]
```diff
@@ -33,11 +33,19 @@
 Then('I can see the welcome message') do
   expect(@test_site.home).to have_welcome_message
 
-  expect(@test_site.home.welcome_message.text).to eq('This is the home page, there is some stuff on it')
+  expect(
```
Comment (luke-hill) [inline features/step_definitions/element_steps.rb]: can we get this onto 2 lines by doing 
```
expect(blah)
  .to eq(blah)
```

As that's the preffered way of doing multilines with multichains. For other items in here also.

File: features/step_definitions/element_steps.rb [hunk 2]
```diff
@@ -76,7 +86,9 @@
   expect(@test_site.home.retiring_element).not_to be_visible
 end
 
-Then('I do not wait for an nonexistent element when checking for invisibility') do
+Then(
+  'I do not wait for an nonexistent element when checking for invisibility'
```
Comment (luke-hill) [inline features/step_definitions/element_steps.rb]: Lets alter this cuke name to something not as silly sounding.

Something like `'I stop waiting once the element is invisible'`

File: features/step_definitions/elements_steps.rb [hunk 3]
```diff
@@ -26,10 +26,17 @@
...
 
-Then('I can wait a variable time for elements to disappear and pass specific parameters') do
-  expect(@test_site.home.wait_for_no_removing_links(0.1, text: 'wibble')).to be true
+Then(
+  "I can wait a variable time for elements to disappear \
+and pass specific parameters"
+) do
+  expect(
+    @test_site.home.wait_for_no_removing_links(0.1, text: 'wibble')
+  ).to be_truthy
```
Comment (luke-hill) [inline features/step_definitions/elements_steps.rb]: This should probably be `not_to raise_error`

File: features/step_definitions/section_steps.rb [hunk 4]
```diff
@@ -61,23 +65,29 @@
...
 end
 
 Then('the section is visible') do
   expect(@test_site.home.people).to be_visible
 end
 
 Then('I can access the sections root element') do
-  expect(@test_site.home.people.root_element.class).to eq(Capybara::Node::Element)
+  expect(
+    @test_site.home.people.root_element.class
```
Comment (luke-hill) [inline features/step_definitions/section_steps.rb]: lets change this expectation to `to be_a CNE`

File: features/step_definitions/sections_steps.rb [hunk 5]
```diff
@@ -1,21 +1,27 @@
 # frozen_string_literal: true
 
 Then('I can see a collection of sections') do
-  @test_site.section_experiments.search_results.each_with_index do |search_result, i|
-    expect(search_result.title.text).to eq("title #{i}")
+  @test_site
```
Comment (luke-hill) [inline features/step_definitions/sections_steps.rb]: For now can we not do this, I think it looks messy. Is there something else we could do? Maybe assigning out the section beforehand to a local variable?

File: features/step_definitions/sections_steps.rb [hunk 6]
```diff
@@ -1,21 +1,27 @@
...
+      expect(search_result.description.text).to eq("description #{i}")
+    end
 
   expect(@test_site.section_experiments.search_results.size).to eq(4)
 end
 
 Then('I can see a collection of anonymous sections') do
-  @test_site.section_experiments.anonymous_sections.each_with_index do |section, index|
-    expect(section.title.text).to eq("Section #{index}")
+  @test_site
```
Comment (luke-hill) [inline features/step_definitions/sections_steps.rb]: same as above?

File: features/step_definitions/wait_steps.rb [hunk 7]
```diff
@@ -53,30 +59,57 @@
 end
 
 When('I wait for the section element that takes a while to disappear') do
-  @test_site.section_experiments.removing_parent_section.wait_for_no_removing_section_element
+  @test_site
```
Comment (luke-hill) [inline features/step_definitions/wait_steps.rb]: lets put this onto 2 lines. `@test_site.section_experiments` then chain the other 2 on line 2

File: features/step_definitions/wait_steps.rb [hunk 8]
```diff
@@ -53,30 +59,57 @@
...
-end
-
-Then(/^an exception is raised when I wait for a section that won't disappear$/) do
-  expect { @test_site.section_experiments.removing_parent_section.wait_for_no_removing_section_element(1) }
-    .to raise_error(SitePrism::TimeOutWaitingForNonExistenceError)
-    .with_message('Timed out after 1s waiting for no Parent#removing_section_element')
-end
-
-When(/^I wait for the collection of sections that takes a while to disappear$/) do
+  expect do
```
Comment (luke-hill) [inline features/step_definitions/wait_steps.rb]: This is way too confusing too read. If we can't fix this up can we just revert this piece and worry about tackling it later.

File: features/step_definitions/wait_steps.rb [hunk 9]
```diff
@@ -53,30 +59,57 @@
...
+      .with_message(
+        'Timed out after 1s waiting for Parent#slow_section_element'
+      )
+  )
+end
+
+Then(
+  /^an exception is raised when I wait for a section that won't disappear$/
+) do
+  expect do
```
Comment (luke-hill) [inline features/step_definitions/wait_steps.rb]: Same here. Also whilst you're here can you convert the regex syntax to string syntax

File: features/step_definitions/wait_steps.rb [hunk 10]
```diff
@@ -53,30 +59,57 @@
...
+  end.to(
+    raise_error(SitePrism::TimeOutWaitingForNonExistenceError)
+      .with_message(
+        'Timed out after 1s waiting for no Parent#removing_section_element'
+      )
+  )
+end
+
+When(
+  /^I wait for the collection of sections that takes a while to disappear$/
```
Comment (luke-hill) [inline features/step_definitions/wait_steps.rb]: This probably needs a rename. Not sure what to though for now.

File: features/step_definitions/wait_steps.rb [hunk 11]
```diff
@@ -103,19 +136,25 @@
   @test_site.home.wait_until_some_slow_element_visible
 end
 
-When('I wait for a specific amount of time until a particular element is visible') do
+When(
+  'I wait for a specific amount of time until a particular element is visible'
```
Comment (luke-hill) [inline features/step_definitions/wait_steps.rb]: `I wait a short amount of time until an element is visible`

File: features/step_definitions/wait_steps.rb [hunk 12]
```diff
@@ -103,19 +136,25 @@
...
   @test_site.home.wait_until_shy_element_visible(5)
 end
 
 When('I wait for an element to become invisible') do
   @test_site.home.wait_until_retiring_element_invisible
 end
 
-When('I wait for a specific amount of time until a particular element is invisible') do
+When(
+  'I wait for a specific amount of time until a particular element is invisible'
```
Comment (luke-hill) [inline features/step_definitions/wait_steps.rb]: `I wait a short amount of time until an element is invisible`

File: test_site/pages/dynamic_page.rb [hunk 13]
```diff
@@ -2,5 +2,5 @@
 
 class TestDynamicPage < SitePrism::Page
   set_url '/dynamic{/letter}.htm'
-  set_url_matcher(/dynamic\/[ab]\.htm$/)
+  set_url_matcher(%r{dynamic\/[ab]\.htm$})
```
Comment (luke-hill) [inline test_site/pages/dynamic_page.rb]: I disagree with this cop. I actually think `//` looks better as it is what is used in many things. Is there a reason rubocop is enforcing this>?

File: features/step_definitions/element_steps.rb [hunk 14]
```diff
@@ -33,11 +33,16 @@
...
   expect(@test_site.home).to have_welcome_message
 
-  expect(@test_site.home.welcome_message.text).to eq('This is the home page, there is some stuff on it')
+  expect(@test_site.home.welcome_message.text)
+    .to eq('This is the home page, there is some stuff on it')
 end
 
 Then('I can see a message using a capybara text query') do
-  expect(@test_site.home).to have_welcome_messages(text: 'This is the home page, there is some stuff on it')
+  expect(@test_site.home).to(
```
Comment (luke-hill) [inline features/step_definitions/element_steps.rb]: Can we get this spec looking like the others

File: features/step_definitions/element_steps.rb (hunk 2 above)
Comment (luke-hill) [reply features/step_definitions/element_steps.rb]: Can we tackle this here?

File: features/step_definitions/element_steps.rb [hunk 15]
```diff
@@ -87,13 +95,18 @@
   @test_site.home.remove_container_with_element_btn.click
 end
 
-Then('I receive an error when a section with the element I am waiting for is removed') do
-  expect { @test_site.home.container_with_element.wait_until_embedded_element_invisible }
-    .to raise_error(Capybara::ElementNotFound)
+Then(
```
Comment (luke-hill) [inline features/step_definitions/element_steps.rb]: I don't like this one atm. Can we just revert it and have a re-think later (Leaving rubocop rules inplace if needs be)

File: features/step_definitions/elements_steps.rb [hunk 16]
```diff
@@ -26,10 +26,16 @@
...
     # intentionally wait and pass nil to force this to cycle
-    expect(@test_site.home.wait_for_lots_of_links(nil, count: 198_108_14)).to be false
+    expect(@test_site.home.wait_for_lots_of_links(nil, count: 198_108_14))
+      .to be_falsey
   end
 end
 
-Then('I can wait a variable time for elements to disappear and pass specific parameters') do
-  expect(@test_site.home.wait_for_no_removing_links(0.1, text: 'wibble')).to be true
+Then(
```
Comment (luke-hill) [inline features/step_definitions/elements_steps.rb]: Same for this one, lets just change it to how it was and tackle it later. I think this format looks off

File: features/step_definitions/page_steps.rb [hunk 17]
```diff
@@ -1,11 +1,15 @@
 # frozen_string_literal: true
 
 Then('I can see an expected bit of the html') do
-  expect(@test_site.home.html).to include('<span class="welcome">This is the home page')
+  expect(
```
Comment (luke-hill) [inline features/step_definitions/page_steps.rb]: Can we alter this spec to look like the 2line ones at the top

File: features/step_definitions/page_steps.rb [hunk 18]
```diff
@@ -1,11 +1,15 @@
...
+    @test_site.home.html
+  ).to include('<span class="welcome">This is the home page')
 end
 
 Then('I can see an expected bit of text') do
-  expect(@test_site.home.text).to include('This is the home page, there is some stuff on it')
+  expect(
```
Comment (luke-hill) [inline features/step_definitions/page_steps.rb]: This one also.

File: features/step_definitions/sections_steps.rb [hunk 19]
```diff
@@ -11,7 +12,9 @@
 end
 
 Then('I can see a collection of anonymous sections') do
-  @test_site.section_experiments.anonymous_sections.each_with_index do |section, index|
+  sections = @test_site.section_experiments.anonymous_sections
+
```
Comment (luke-hill) [inline features/step_definitions/sections_steps.rb]: lets delete this newline for consistency

File: features/step_definitions/wait_steps.rb [hunk 20]
```diff
@@ -71,30 +77,45 @@
...
-  expect(@test_site.section_experiments.removing_parent_section).not_to have_removing_section_element
+  expect(@test_site.section_experiments.removing_parent_section)
+    .not_to have_removing_section_element
 end
 
 Then(/^an exception is raised when I wait for a section that won't appear$/) do
-  expect { @test_site.section_experiments.parent_section.wait_for_slow_section_element(1) }
-    .to raise_error(SitePrism::TimeOutWaitingForExistenceError)
+  parent = @test_site.section_experiments.parent_section
+  match_matcher = \
```
Comment (luke-hill) [inline features/step_definitions/wait_steps.rb]: Can we revert this one for now and tackle it later

File: features/step_definitions/wait_steps.rb [hunk 21]
```diff
@@ -71,30 +77,45 @@
...
     .with_message('Timed out after 1s waiting for Parent#slow_section_element')
+
+  expect { parent.wait_for_slow_section_element(1) }.to match_matcher
 end
 
-Then(/^an exception is raised when I wait for a section that won't disappear$/) do
-  expect { @test_site.section_experiments.removing_parent_section.wait_for_no_removing_section_element(1) }
-    .to raise_error(SitePrism::TimeOutWaitingForNonExistenceError)
-    .with_message('Timed out after 1s waiting for no Parent#removing_section_element')
+Then(
```
Comment (luke-hill) [inline features/step_definitions/wait_steps.rb]: Same for this one, I think we need to think how we want these to read.

File: features/step_definitions/wait_steps.rb (hunk 11 above)
Comment (luke-hill) [reply features/step_definitions/wait_steps.rb]: Can we fix this cuke up?

File: features/step_definitions/wait_steps.rb (hunk 12 above)
Comment (luke-hill) [reply features/step_definitions/wait_steps.rb]: And this one

File: test_site/sections/search_result.rb [hunk 22]
```diff
@@ -6,7 +6,10 @@ class SearchResult < SitePrism::Section
 
   def cell_value=(value)
     execute_script(
-      "document.getElementById('first_search_result').children[0].innerHTML = '#{value}'"
+      <<-TEXT
```
Comment (luke-hill) [inline test_site/sections/search_result.rb]: Is this better? I'm not so sure?

File: .rubocop.yml [hunk 23]
```diff
@@ -25,6 +25,3 @@ Style/RegexpLiteral:
 BlockLength:
   Exclude:
     - './spec/*_spec.rb'
-
-Gemspec/RequiredRubyVersion:
```
Comment (luke-hill) [inline .rubocop.yml]: We need to re-add this for now (Whilst we have legacy ruby versions supported)

File: features/step_definitions/sections_steps.rb [hunk 24]
```diff
@@ -1,19 +1,19 @@
 # frozen_string_literal: true
 
 Then('I can see a collection of sections') do
-  @test_site.section_experiments.search_results.each_with_index do |search_result, i|
+  results = @test_site.section_experiments.search_results
+  results.each_with_index do |search_result, i|
     expect(search_result.title.text).to eq("title #{i}")
-
```
Comment (luke-hill) [inline features/step_definitions/sections_steps.rb]: can we leave line-spaces inbetween dual expectations. In spec below also

File: .rubocop.yml (hunk 23 above)
Comment (ineverov) [reply .rubocop.yml]: ah ok

**Pull request merged**